File: midterm-Lab3/MP18/MP18_RemoveEmptyRows.java
import java.io.*;
import java.util.*;

/*
MP18 – Remove rows with empty fields
This program reads a CSV dataset and displays only rows
that do not contain empty fields.
*/

public class MP18_RemoveEmptyRows {

    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);

        // Ask user for dataset file path
        System.out.print("Enter dataset file path: ");
        String filePath = input.nextLine();

        try {

            BufferedReader reader = new BufferedReader(new FileReader(filePath));
            String line;

            System.out.println("\nRows without empty fields:\n");

            // Read each row of the CSV
            while ((line = reader.readLine()) != null) {

                // Split row into columns
                String[] columns = line.split(",");

                boolean hasEmptyField = false;

                // Check if any column is empty
                for (String col : columns) {

                    if (col.trim().isEmpty()) {
                        hasEmptyField = true;
                        break;
                    }

                }

                // Display row only if it has no empty fields
                if (!hasEmptyField) {
                    System.out.println(line);
                }

            }

            reader.close();

        } catch (Exception e) {

            System.out.println("Error reading file.");

        }

    }
}

File: midterm-Lab3/MP18/MP18_RemoveEmptyRows.js
/*
MP18 – Remove rows with empty fields
This program reads a CSV dataset and displays only rows
that do not contain any empty fields.
*/

const fs = require("fs");
const readline = require("readline");

// Create input interface
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

// Ask user for dataset file path
rl.question("Enter dataset file path: ", function(path) {

    try {

        // Read the file
        const data = fs.readFileSync(path, "utf8");

        // Split rows and remove extra line endings
        const rows = data.split("\n").map(row => row.trimEnd());

        console.log("\nRows without empty fields:\n");

        rows.forEach(row => {

            if (row.trim() === "") {
                return;
            }

            const columns = row.split(",");

            let hasEmptyField = false;

            // Check each column
            for (let col of columns) {
                if (col.trim() === "") {
                    hasEmptyField = true;
                    break;
                }
            }

            // Print row if no empty fields
            if (!hasEmptyField) {
                console.log(row);
            }

        });

    } catch (err) {

        console.log("Error reading file.");

    }

    rl.close();

});